<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="arm_left"/>
  <arg name="arm_right"/>
  <arg name="end_effector_left"/>
  <arg name="end_effector_right"/>
  <arg name="ft_sensor_left"/>
  <arg name="ft_sensor_right"/>
  
<arg name="file_suffix" value="$(eval 
              (('no-arm-left' if not arm_left else (end_effector_left if end_effector_left else 'no-ee') + ('_' +  ft_sensor_left if ft_sensor_left else '')) + '_'  + ('no-arm-right' if not arm_right else (end_effector_right  if end_effector_right else 'no-ee') + ('_' +  ft_sensor_right if ft_sensor_right else ''))))"/>
  <!-- execute pre-recorded motions -->
  <rosparam file="$(find tiago_dual_bringup)/config/motions/tiago_motions_general.yaml" command="load" if="$(eval (arm_right and arm_left))"/> 
  <rosparam file="$(find tiago_dual_bringup)/config/motions/tiago_motions_general_arm_right.yaml" command="load" if="$(eval (arm_right and not arm_left))"/> 

   <rosparam file="$(find tiago_dual_bringup)/config/motions/tiago_motions_general_arm_left.yaml" command="load" if="$(eval (arm_left and not arm_right))"/> 


  <!-- any end-effector motions -->
  <rosparam unless="$(eval (end_effector_left == 'custom' and end_effector_right == 'custom') or (end_effector_left == 'custom' and not arm_right) or (not arm_left and end_effector_right == 'custom') )" file="$(find tiago_dual_bringup)/config/motions/tiago_motions_$(arg file_suffix).yaml" command="load" />
  <rosparam if="$(eval end_effector_left == 'custom' or end_effector_right == 'custom')" file="$(find custom_dual_ee_configuration)/config/motions/tiago_dual_custom_motions.yaml" command="load" />
  <node pkg="play_motion" type="play_motion" name="play_motion">
    <rosparam if="$(eval end_effector_left != 'custom' and end_effector_right != 'custom')" file="$(find tiago_dual_bringup)/config/approach_planner/approach_planner_$(arg file_suffix).yaml" command="load" />
    <rosparam if="$(eval end_effector_left == 'custom' or end_effector_right == 'custom')" file="$(find custom_dual_ee_configuration)/config/approach_planner/approach_planner.yaml" command="load" />
  </node>

  <!-- I don't think this is being used -->
  <!--<node pkg="play_motion" type="is_already_there.py" name="is_already_there" />-->
</launch>
